src/components/KCodeBlock/KCodeBlock.vue

```diff
@@ -135,66 +136,16 @@
       </div>
     </div>
 
-    <div class="code-block-content">
-      <!-- eslint-disable vue/no-v-html -->
-      <pre
-        v-if="isShowingFilteredCode"
-        class="filtered-code-block"
-        data-testid="filtered-code-block"
-      >
-        <span
-          v-if="!singleLine"
-          class="line-number-rows"
-        >
-          <span
-            v-for="line in matchingLineNumbers"
-            :key="line"
-            class="line"
-          >
-            <a
-              :id="`${linePrefix}-L${line}`"
-              class="line-anchor"
-              :href="showLineNumberLinks ? `#${linePrefix}-L${line}` : undefined"
-            >{{ line }}</a>
-          </span>
-        </span>
-        <code v-html="filteredCode" />
-      </pre>
-
-      <pre
-        v-else
-        class="highlighted-code-block"
-        :class="{
-          'single-line': singleLine,
-          'show-copy-button': showCopyButton
-        }"
-        data-testid="highlighted-code-block"
-      >
-        <span
-          v-if="!singleLine"
-          class="line-number-rows"
-        >
-          <span
-            v-for="line in totalLines"
-            :key="line"
-            class="line"
-            :class="{
-              'line-is-match': matchingLineSet.has(line),
-              'line-is-highlighted-match': currentLineIndex !== null && line === matchingLineNumbers[currentLineIndex],
-            }"
-          >
-            <a
-              :id="`${linePrefix}-L${line}`"
-              class="line-anchor"
-              :class="{ 'hide-links': !showLineNumberLinks }"
-              :href="showLineNumberLinks ? `#${linePrefix}-L${line}` : undefined"
-            >{{ line }}</a>
-          </span>
-        </span>
-        <code v-html="finalCode" />
-      </pre>
-      <!-- eslint-enable vue/no-v-html -->
-
+    <div
+      ref="codeBlockContent"
+      class="code-block-content"
+    >
+      <!--
+        This was moved up before the lines because 1Password's (yes, 1Password!) heuristics will cause a delay because
+        it will walk through all line number elements one by one to calculate layout metrics, this will cause the
+        browser to schedule layout recalculations after each element is processed.
+        Though we are using virtual scrolling now, this still helps with the overall performance.
+      -->
```

Comment:
nit: we need to remove these inline HTML comments before merge; we don't want to ship these. If we want the record, you can add block comments in the script section, etc.